modules/dreamview/backend/handlers/websocket.cc

```diff
@@ -144,29 +144,50 @@ bool WebSocketHandler::SendData(Connection *conn, const std::string &data,
   return true;
 }
 
+thread_local unsigned char WebSocketHandler::current_opcode_ = 0x00;
+thread_local std::stringstream WebSocketHandler::data_;
+
 bool WebSocketHandler::handleData(CivetServer *server, Connection *conn,
                                   int bits, char *data, size_t data_len) {
   // Ignore connection close request.
   if ((bits & 0x0F) == WEBSOCKET_OPCODE_CONNECTION_CLOSE) {
     return false;
   }
 
-  switch (bits &= 0x7f) {
-    case WEBSOCKET_OPCODE_TEXT:
-      return handleJsonData(conn, data, data_len);
-    case WEBSOCKET_OPCODE_BINARY:
-      return handleBinaryData(conn, data, data_len);
-    default:
-      AERROR << "Unknown WebSocket bits flag: " << bits;
-      return true;
+  WebSocketHandler::data_.write(data, data_len);
+  if (WebSocketHandler::current_opcode_ == 0x00) {
+    WebSocketHandler::current_opcode_ = bits & 0x7f;
   }
+
+  bool result = true;
+  bool isFinalFragment = bits & 0x80;
+  if (isFinalFragment) {
+    switch (WebSocketHandler::current_opcode_) {
+      case WEBSOCKET_OPCODE_TEXT:
+        result = handleJsonData(conn, WebSocketHandler::data_.str());
+        break;
+      case WEBSOCKET_OPCODE_BINARY:
+        result = handleBinaryData(conn, WebSocketHandler::data_.str());
+        break;
+      default:
+        AERROR << "Unknown WebSocket bits flag: " << bits;
+        break;
+    }
+
+    // rest opcode and data
+    WebSocketHandler::current_opcode_ = 0x00;
+    WebSocketHandler::data_.clear();
+    WebSocketHandler::data_.str(std::string());
```

Comment:
I guess just .clear() is enough? And data_ instead of WebSocketHandler::data_?

Reply:
.clear() is used to clear the error state of the stream. Needs str(std::string()) to empty the stream.